### ISSUE TITLE
Ignore `NSMenu` window

### ISSUE THREAD

**Issue by sindresorhus** [sbmpost/AutoRaise#82]
AutoRaise causes problems with menu bar apps that use NSMenu since it tries to activate the app. When a NSMenu is open and the app gets activated, the NSMenu closes immediately.

I'm not familiar with the AX APIs, but you may be able to detect that it's a `NSMenu` window. The class name for a `NSMenu` window is `NSMenuWindowManagerWindow`.

**Comment by sbmpost** [comment]
@sindresorhus 
Thanks for creating the issue. Do you have any example apps I can test with?

**Comment by sindresorhus** [comment]
You can run the trial version of my Dato app at: https://sindresorhus.com/dato#trial

**Comment by sbmpost** [comment]
@sindresorhus 

> For example the UI elements in the "Dock & Menu bar" window in "System Preferences" sometimes point to a parent window that has no title

After writing this down, I figured how is this even possible? As it turns out, giving priority over uit parent windows before considering it to be a window itself makes things more consistent. In turn this enables me to -always- check for windows without a title which should solve your use case. I pushed these changes to [this branch](https://github.com/sbmpost/AutoRaise/tree/check-for-non-focusable-windows). Note however that eventhough the changes are minor, the change in behaviour might be quite substantial. So I will be running this for at least a couple of weeks to see if it doesn't break other corner cases. So far so good however.
